app/models/spree/gateway/stripe_sca.rb

```diff
@@ -78,7 +82,13 @@ def void(response_code, _creditcard, gateway_options)
         payment_intent_response = Stripe::PaymentIntent.retrieve(payment_intent_id,
                                                                  stripe_account: stripe_account_id)
         gateway_options[:stripe_account] = stripe_account_id
-        provider.refund(refundable_amount(payment_intent_response), response_code, gateway_options)
+
+        # If a payment has been confirmed it cannot be voided by Stripe, and must be refunded instead
+        if voidable?(payment_intent_response)
+          provider.void(response_code, gateway_options)
+        else
+          provider.refund(refundable_amount(payment_intent_response), response_code, gateway_options)
+        end
```

Comment:
Re-review note: this was the weird bit. 
- `StripeSCA#purchase` calls `provider#purchase` :heavy_check_mark: 
- `StripeSCA#capture` calls `provider#capture` :heavy_check_mark: 
- `StripeSCA#void` calls... `provider#refund`! What!?

Reply:
This was changed a while ago. The issue is that we can't void (cancel) payments that are confirmed/completed. But that change meant that we also couldn't void incomplete payments at all :see_no_evil:

Reply:
Documentation for reference: https://stripe.com/docs/api/payment_intents/cancel (the void method via `ActiveMerchant` calls this endpoint)